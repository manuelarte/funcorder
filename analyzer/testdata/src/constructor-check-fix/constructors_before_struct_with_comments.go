package constructor_check_fix

// NewOtherConstructorAfterStructMethodsWithComments This constructor creates the
// struct ConstructorBeforeStructWithComments
// with a named return
func NewOtherConstructorBeforeStructWithComments() (m *ConstructorBeforeStructWithComments) { // want `constructor \"NewOtherConstructorBeforeStructWithComments\" for struct \"ConstructorBeforeStructWithComments\" should be placed after the struct declaration`
	m = &ConstructorBeforeStructWithComments{Name: "John"}
	return
}

func NewConstructorBeforeStructWithComments() *ConstructorBeforeStructWithComments { // want `constructor \"NewConstructorBeforeStructWithComments\" for struct \"ConstructorBeforeStructWithComments\" should be placed after the struct declaration`
	return &ConstructorBeforeStructWithComments{Name: "John"}
}

type ConstructorBeforeStructWithComments struct {
	Name string
}

// GetName returns the name.
func (m ConstructorBeforeStructWithComments) GetName() string {
	return m.Name
}

// SetName sets the name
// multi line comment
func (m *ConstructorBeforeStructWithComments) SetName(name string) {
	m.Name = name
}
