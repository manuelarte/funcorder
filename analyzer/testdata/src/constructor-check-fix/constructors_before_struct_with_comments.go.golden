package constructor_check_fix

type ConstructorBeforeStructWithComments struct {
	Name string
}

// NewOtherConstructorAfterStructMethodsWithComments This constructor creates the
// struct ConstructorBeforeStructWithComments
// with a named return
func NewOtherConstructorBeforeStructWithComments() (m *ConstructorBeforeStructWithComments) {
	m = &ConstructorBeforeStructWithComments{Name: "John"}
	return
}

func NewConstructorBeforeStructWithComments() *ConstructorBeforeStructWithComments {
	return &ConstructorBeforeStructWithComments{Name: "John"}
}

// GetName returns the name.
func (m ConstructorBeforeStructWithComments) GetName() string {
	return m.Name
}

// SetName sets the name
// multi line comment
func (m *ConstructorBeforeStructWithComments) SetName(name string) {
	m.Name = name
}
